zerver/tests/test_messages.py

```diff
@@ -3204,6 +3206,167 @@ def test_propagate_all_topics(self) -> None:
         self.check_message(id5, topic_name="edited")
         self.check_message(id6, topic_name="topic3")
 
+    def prepare_move_topics(self, user_email: str, old_stream: str, new_stream: str, topic: str) -> Tuple[UserProfile, Stream, Stream, int]:
+        user_profile = self.example_user(user_email)
+        self.login(user_profile.email)
+        stream = self.make_stream(old_stream)
+        new_stream = self.make_stream(new_stream)
+        self.subscribe(user_profile, stream.name)
+        self.subscribe(user_profile, new_stream.name)
+        msg_id = self.send_stream_message(user_profile, stream.name,
+                                          topic_name=topic, content="First")
+        self.send_stream_message(user_profile, stream.name,
+                                 topic_name=topic, content="Second")
+
+        self.send_stream_message(user_profile, stream.name,
+                                 topic_name=topic, content="third")
+
+        return (user_profile, stream, new_stream, msg_id)
+
+    def test_move_message_to_stream(self) -> None:
+        prepared_topics = self.prepare_move_topics("iago", "test move stream",
+                                                   "new stream", "test")
+        user_profile = prepared_topics[0]
+        stream = prepared_topics[1]
+        new_stream = prepared_topics[2]
+        msg_id = prepared_topics[3]
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
+            'message_id': msg_id,
+            'new_stream_id': new_stream.id,
+            'propagate_mode': 'change_all'
+        })
+
+        messages = get_topic_messages(user_profile, stream, "test")
+        self.assertEqual(len(messages), 1)
+        self.assertEqual(messages[0].content, "This topic was moved by @**Iago|10** to #**new stream>test**")
+
+        messages = get_topic_messages(user_profile, new_stream, "test")
+        self.assertEqual(len(messages), 4)
+        self.assertEqual(messages[3].content, "This topic was moved here from #**test move stream>test** by @**Iago|10**")
+
+        self.assert_json_success(result)
+
+    def test_move_message_to_stream_change_later(self) -> None:
+        prepared_topics = self.prepare_move_topics("iago", "test move stream",
+                                                   "new stream", "test")
+        user_profile = prepared_topics[0]
+        stream = prepared_topics[1]
+        new_stream = prepared_topics[2]
+        msg_id = prepared_topics[3]
+
+        result = self.client_patch("/json/messages/" + str(msg_id), {
```

Comment:
Please use `msg_id + 1` here; we want to verify that it only moved some of the messages.  Otherwise, this isn't a very good test of the `change_later` feature!

Reply:
Sorry, I don't catch it. What is the expected result for this test?? What do you mean by using `msg_id + 1`?

Reply:
I mean move the 2nd message and later, not what you have moving the first message and later (which ends up being all the messages and thus doesn't test the precise behavior of change_later).